Content.Server/Sports/BaseballBatSystem.cs

```diff
@@ -0,0 +1,85 @@
+using Content.Shared.Throwing;
+using Content.Shared.Weapons.Melee;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Content.Shared.Weapons.Ranged;
+using Content.Shared.Weapons.Ranged.Systems;
+using Robust.Shared.Timing;
+
+
+namespace Content.Server.Sports
+{
+    public sealed class BaseballBatSystem : EntitySystem
+    {
+        [Dependency] private readonly ThrowingSystem _throwingSystem = default!;
+        [Dependency] private readonly EntityLookupSystem _entityLookupSystem = default!;
+        [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly GunSystem _gunSystem = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<BaseballBatComponent, WideAttackEvent>(OnBatAttack);
+        }
+
+        public void OnBatAttack(EntityUid uid, BaseballBatComponent component, WideAttackEvent args)
+        {
+            args.Handled = true;
+
+            if (!EntityManager.TryGetComponent<MeleeWeaponComponent>(component.Owner, out var meleeWeaponComponent))
+                return;
+            if (_gameTiming.CurTime < meleeWeaponComponent.CooldownEnd)
+                return;
+
+            var location = EntityManager.GetComponent<TransformComponent>(args.User).Coordinates;
+            var dir = args.ClickLocation.ToMapPos(EntityManager) - location.ToMapPos(EntityManager);
+            var dirForceMultiplier = _random.NextFloat(component.WackForceMultiplierMin, component.WackForceMultiplierMax);
+
+            var hitStrength = _random.NextFloat(component.WackStrengthMin, component.WackStrengthMax);
+
+            var transformQuery = GetEntityQuery<TransformComponent>();
+            var physicsQuery = GetEntityQuery<PhysicsComponent>();
+            var itemQuery = GetEntityQuery<ItemComponent>();
+            var thrownItemQuery = GetEntityQuery<ThrownItemComponent>();
+
+
+            //The melee system uses a collision raycast but apparently that doesnt work with items so im using GetEntitiesInArc
+            foreach (var entity in _entityLookupSystem.GetEntitiesInArc(location, meleeWeaponComponent.Range, dir.ToAngle(), meleeWeaponComponent.ArcWidth))
+            {
+                //Checking to see if the items are actually throwable
+                if (!itemQuery.HasComponent(entity))
+                    return;
+                if (transformQuery.TryGetComponent(entity, out var transformComponent) && transformComponent.Anchored)
+                    return;
+                if (!physicsQuery.TryGetComponent(entity, out var physicsComponent))
+                    return;
+                if (!thrownItemQuery.HasComponent(entity) && component.OnlyHitThrown)
+                    continue;
```

Comment:
This looks very expensive for no good reason, cant you just check if `entity` has `BallTagComponent` and if you want, check if its `BodyStatus` is `InAir` if you only want this system to work when the ball is thrown (though pitching a stationary one shouldn't be an issue) instead of these queries?

Reply:
The idea is that the bat can hit any _thrown_ item, I experimented with checking bodystatus but I don't want players to be able to smack bullets out of the air

Reply:
I'm trimmed it down to just checking if it has thrownitemcomponent and removed the OnlyHitThrown feature